appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -629,6 +719,191 @@ protected String getRwString() {
     }
   }
 
+  /**
+   * An enum specifying the available types of helper blocks aka types of helper UI.
+   */
+  protected enum HelperType { OPTION_LIST }
+
+  /**
+   * A key that allows you to access info about a helper block.
+   */
+  protected final class HelperKey {
+    private HelperType helperType;
+
+    private String key;
+
+    /**
+     * Creates a HelperKey which can be used to access data about a helper block.
+     */
+    protected HelperKey(HelperType type, String key) {
+      this.helperType = type;
+      this.key = key;
+    }
+
+    /**
+     * Returns the type of helper block, aka the type of helper UI. Eg an option list.
+     * @return the type of helper block.
+     */
+    protected HelperType getType() {
+      return helperType;
+    }
+
+    /**
+     * Returns the key to the specific helper data. Eg in the case of an option list helper, this
+     * key could be used to look up values in the optionLists Map.
+     * @return key to the helper data.
+     */
+    protected String getKey() {
+      return key;
+    }
+  }
+
+
+  /**
+   * Represents a list of Options associated with some (enum) class. The data in this OptionList
+   * is used to create OptionList helper blocks. 
+   */
+  protected final class OptionList {
+    /**
+     * A list of option values (Strings) and option info (Options).
+     * For built-in components the Option name is used to look up the translated display text.
+     * For extensions, which do not support i18n, the Option name /is/ the display text.
+     */
+    private ArrayList<Option> options;
+
+    /**
+     * The fully qualified class name this OptionList is associated with.
+     */
+    private String className;
+
+    /**
+     * The tag name this OptionList is associated with. This goes in front of the dropdown in the
+     * blocks editor. It is usually the simplified class name.
+     */
+    private String tagName;
+
+    /**
+     * The Option.name of the default option associated with this OptionList.
+     */
+    private String defaultOpt;
+
+    /**
+     * Creates an OptionList (which is a definition of a option list helper-block) that can be
+     * populated with options.
+     * 
+     * @param className The fully qualified class name this OptionList is associated with.
+     * @param tagName The tag name this OptionList is associated with. This goes in front of the
+     *     dropdown in the blocks editor. It is usually the simplified class name.
+     */
+    protected OptionList(String className, String tagName) {
+      this.className = className;
+      this.tagName = tagName;
+      options = new ArrayList();
+    }
```

Comment:
Currently it doesn't collect them because that info isn't being used anywhere. The dropdown block always returns an abstract type, so it doesn't need to know about the concrete type. And block inputs get the concrete type from the parameter types, not from the OptionList info.

So yeah! It would be pretty easy to grab the concrete type info if we wanted to grab it, and that might make the system more robust. I just couldn't think of a use case for it at the time hehe.

Reply:
[Done!](https://github.com/BeksOmega/appinventor-sources/pull/3/commits/79bca2685edb0a2ecbf44aee767c358fa185bb7e)